pandas/core/array_algos/take.py

```diff
@@ -57,20 +57,29 @@ def take_nd(
 
     Returns
     -------
-    subarray : array-like
+    subarray : np.ndarray or ExtensionArray
         May be the same type as the input, or cast to an ndarray.
     """
     if fill_value is lib.no_default:
         fill_value = na_value_for_dtype(arr.dtype, compat=False)
 
-    arr = extract_array(arr, extract_numpy=True)
-
     if not isinstance(arr, np.ndarray):
         # i.e. ExtensionArray,
         # includes for EA to catch DatetimeArray, TimedeltaArray
         return arr.take(indexer, fill_value=fill_value, allow_fill=allow_fill)
 
     arr = np.asarray(arr)
+    return _take_nd_ndarray(arr, indexer, axis, out, fill_value, allow_fill)
+
+
+def _take_nd_ndarray(
```

Comment:
this is purely for typing right?

Reply:
yah mypy complained without it

Reply:
you can't just cast here?

Reply:
simon explained a couple days ago in a different thread that it involves ArrayLike being a TypeVar and not a union

Reply:
ok then